v2/pkg/operators/common/dsl/dsl.go

```diff
@@ -490,6 +492,32 @@ func init() {
 			} else if govalidator.IsFloat(argStr) {
 				sint, err := strconv.ParseFloat(argStr, 64)
 				return float64(sint), err
+			} else if govalidator.IsIP(argStr) { // add by 51pwn
+				ip := net.ParseIP(argStr)
+				i := big.NewInt(0)
+				i.SetBytes(ip)
+				return i.Int64(), nil
+			} else if govalidator.IsDNSName(argStr) { // add by 51pwn
```

Comment:
What about adding two independent helper functions called for example `dns_to_ip` and `ip_to_number`. Could you also provide a bit more context about the use cases?

Reply:
yes，thanks

Reply:
The helper functions seem to be very specific. Would it be possible to provide more context about their use cases? For which scenarios do you need to resolve a domain name to IP and convert it to a number?

Reply:
@Mzack9999
scene

1. Check whether the response contains IP_ to_ number(xx), ip_ to_ Special value for number (dns_to_ip (XX))

2. IP this data_ to_ number(xx), ip_ to_ Number (dns_to_ip (XX)) is implicitly sent to other systems for verification

3. Use the converted number to make a specific payload